Fix tests

gctucker opened the pull request:
This is to fix various issues with the unit tests and make it possible to extend them with new IPC modules.

* Fix false positive failures with the `ZeromqIPC` due to arbitrary sleep; wait for the VSM process to end instead
* Separate `ZeromqIPC` related code to a new class and run the same test cases with both `ZeromqIPC` and `DebugIPC`

gctucker committed:
tests: move ZeroMQ tests to a separate class

Split the TestVSM class into TestVSMDebug which deals with DebugIPC
and TestVSMZeroMQ which deals with ZeromqIPC.

Also split the TestVSM class to separate the run_vsm logic from the
test cases, which are now in VSMTestCases.  Create new sub-classes of
VSMTestCases which defines which IPC module to use.

This removes a lot of conditionals and makes the code easier to
maintain and expand with new IPC modules that need to run a different
set of test cases.

Signed-off-by: Guillaume Tucker <guillaume.tucker@collabora.com>

gctucker committed:
tests: call process.wait() rather than sleep when testing zeromq

Call process.wait() to wait until the VSM has completed, rather than
sleeping for an arbitrary amount of time.  This fixes cases where
delayed signals would not be emitted by the time the VSM process was
terminated.

Signed-off-by: Guillaume Tucker <guillaume.tucker@collabora.com>

kov reviewed (changes requested):
I'm not a big fan of mixins, have you considered putting the tests on a base class? They could be in TestVSM, since it is the base class for the ZeroMQ class, or it can be a separate one that inherits from unittest.TestCase and gets inherited by TestVSM?

gctucker commented:
The issue is that to test the case when an IPC module returns `None` (see #44) I need to run a test case with a test IPC class and not with the regular IPC modules (`DebugIPC`, `ZeromqIPC`) as they don't have the concept of unsupported signals.

One thing I can do then is create an instance of an "IPC manager" class during `setUp()`.  It's going to be a bit clunky but hopefully you'll like it better :)

gctucker commented:
@kov I've reworked this PR and rebased #44 on top of it and added this test case for the `None` signals:
https://github.com/GENIVI/vehicle_signal_manager/pull/44/commits/42ced8cad51c879573a1bd8cd9c899e738c42e13

kov reviewed (approved):
Thanks!
